src/app.ts

```diff
@@ -17,8 +20,13 @@ app.use(logger("dev"));
 app.use(express.static("public"));
 app.use(favicon(__dirname + "/public/favicon.ico"));
 
+
 // Use main router for API routes
 app.use("/api/v1", mainRouter);
 
 
-export default app;
\ No newline at end of file
+app.use(notFoundMiddleware); // 404 Not Found handler
```

Comment:
I had to do a bunch of personal research to understand what you are doing here. I think it would be valuable to include in your commit message how these work ("add a middleware that catches any routes not handled by the mainRouter" for example) and/or a link to a tutorial you are following for these middleware.